packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,78 +4,106 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
 import { useEffect, useRef, useState } from 'react'
 
-const initialValue = 0
-
 let prefersReducedMotion
 const isBrowser = typeof window !== 'undefined'
 if (isBrowser) {
   prefersReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches
 }
 
-function AnimatedNumber({ duration = 1000, value }) {
-  const numRef = useRef(null)
-  const [animated, setAnimated] = useState(false)
+function formatValue(num) {
+  return format(',d')(num)
+}
 
-  function animateValue() {
-    select(numRef.current)
+/**
+ * Use d3 to animate `element` from `initialValue` to `value` over `duration` in milliseconds.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue 
+ * @param {number} value
+ * @param {number} duration 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function animateValue(element, initialValue, value, duration) {
+  if (value === initialValue) {
+    return
+  }
+  return new Promise((resolve) => {
+    select(element)
       .data([value])
       .transition()
       .duration(duration)
       .textTween(() => {
         const interpolator = interpolate(initialValue, value)
         return t => {
           const interpolatedValue = interpolator(t)
-          if (interpolatedValue === value) setAnimated(true) // animation complete!
+          if (interpolatedValue === value) {
+            resolve() // animation complete!
+          }
           const niceValue = formatValue(interpolatedValue)
           return niceValue
         }
       })
-  }
+  })
+}
 
-  function lessAnimation() {
-    select(numRef.current)
-    .data([value])
-    .transition()
-    .duration(0)
-    .textTween(() => {
-      return () => {
-        setAnimated(true) // animation complete!
-        return formatValue(value)
-      }
-    })
+/**
+ * Use d3 to tween `element` from `initialValue` to `value`.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue
+ * @param {number} value 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function lessAnimation(element, initialValue, value) {
+  if (value === initialValue) {
+    return
   }
+  return new Promise((resolve) => {
+    select(element)
+      .data([value])
+      .transition()
+      .duration(0)
+      .textTween(() => {
+        return () => {
+          resolve() // animation complete!
+          return formatValue(value)
+        }
+      })
+  })
+}
 
-  function formatValue(num) {
-    return format(',d')(num)
-  }
+function AnimatedNumber({ duration = 1000, value }) {
+  const numRef = useRef(null)
+  const initialValueRef = useRef(0)
+  const initialValue = initialValueRef.current
+  const [animated, setAnimated] = useState(false)
 
   useEffect(() => {
     // If we already animated the number once, don't observe intersection
     // This could happen if the value prop updates, but page did not refresh
-    if (animated) return
+    if (animated || value === initialValue) return
```

Comment:
One bug in the original code is that it would create the animation callback even if `initialValue` and `value` are both 0. I've fixed that by storing the previous value of `value` in `initialValue`, once the animation completes. The name of `initialValue` might need to be changed, for clarity. I'm not sure.

Reply:
`value` can still continue to increment as we fetch fresh classification statistics from the API, or as local values increment, so `animated` still needs to be checked, if we only want to animate once.

Reply:
Renamed `initialValue` to `displayedValue`, since it is the value that's shown to a volunteer.